File: spoj/classicals/NY10A.py
from collections import defaultdict

seq = ["TTT", "TTH", "THT", "THH", "HTT", "HTH", "HHT", "HHH"]

def solve(a, n):
	store = defaultdict(int)
	cur = ''
	for s in a:
		cur += s
		for t in seq:
			#print store
			if cur == t:

				store[t] += 1
				cur = cur[1:]
	print n,
	for s in seq:
		print store[s],
	print

for _ in xrange(input()):
	n = input()
	a = raw_input()
	solve(a, n)

File: spoj/classicals/NY10A.cpp
#include <map>
#include <iostream>

using namespace std;

string seq[] = {"TTT", "TTH", "THT", "THH", "HTT", "HTH", "HHT", "HHH"};

void solve(string a, int n){
	map<string, int> store;
	string cur = "";
	for (int i = 0; i < a.length(); i += 1){
		string s = a.substr(i,1);
		cur += s;
		for (string x: seq){
			if (cur == x){
				store[x] += 1;
				cur = cur.substr(1,2);
			}
		}
	}
	cout << n << " ";
	for(string x: seq){
		cout << store[x] << " ";
	}
	cout << endl;
}

int main(void){
	int t, n;
	string a;
	cin >> t;
	while (t--){
		cin >> n;
		cin >> a;
		solve(a, n);
	}
}
